src/exts/tvdb_info/ui.py

```diff
@@ -0,0 +1,512 @@
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from typing import NamedTuple, TYPE_CHECKING, TypedDict, override
+
+import discord
+
+from src.bot import Bot
+from src.db_adapters import (
+    get_list_item,
+    list_put_item,
+    list_remove_item,
+    refresh_list_items,
+    user_get_list_safe,
+    user_get_safe,
+)
+from src.db_tables.user_list import UserList, UserListItem, UserListItemKind
+from src.settings import THETVDB_COPYRIGHT_FOOTER, THETVDB_LOGO
+from src.tvdb import Movie, Series
+from src.utils.log import get_logger
+from src.utils.tvdb import by_season
+
+if TYPE_CHECKING:
+    from src.db_tables.user import User
+    from src.tvdb.client import Episode
+
+log = get_logger(__name__)
+
+MOVIE_EMOJI = "🎬"
+SERIES_EMOJI = "📺"
+
+
+class ReactiveButtonState(TypedDict):
+    """Type definition for a reactive button state."""
+
+    label: str
+    style: discord.ButtonStyle
+    emoji: str
+
+
+# A reactive button is a tuple of the button, the item in the list, the state when active, and the state when inactive.
+# It allows us to programmatically change the button's appearance based on the state of the item in the list.
+
+
+class ReactiveButton(NamedTuple):
+    """A tuple of the button, the item in the list, the state when active, and the state when inactive."""
+
+    button: discord.ui.Button  # pyright: ignore[reportMissingTypeArgument]
+    item: UserListItem | None
+    active_state: ReactiveButtonState
+    inactive_state: ReactiveButtonState
+
+
+class _ReactiveView(discord.ui.View, ABC):
+    _reactive_buttons_store: list[ReactiveButton] | None = None
+    bot: Bot
+
+    async def _update_states(self) -> None:
+        for button, item, active_state, inactive_state in await self._reactive_buttons:
```

Comment:
This unpacking essentially removes the benefit of using that named tuple, as it relies on the order. Instead, do `for reactive_button in reactive_buttons:` and access these as named attributes.

Reply:
Actually this can remain, as it does leave the code a bit cleaner to look at.